brave/ui/app/ducks/metamask/metamask.js

```diff
@@ -4,9 +4,85 @@ module.exports = function (state, action) {
   const newState = reduceMetamask(state, action)
   newState.batTokenAdded = newState.batTokenAdded || false
 
-  if (action.type === 'SET_BAT_TOKEN_ADDED') {
-    newState.batTokenAdded = action.value
+  newState.externalProvider = newState.externalProvider || {
+    accounts: {
```

Comment:
all this data is just mock stuff that was used for testing, ultimately it will come from an actual API that background.js interacts with